examples/call/solana/programs/connected/src/lib.rs

```diff
@@ -28,23 +48,91 @@ pub mod connected {
         let message = String::from_utf8(data).map_err(|_| ErrorCode::InvalidDataFormat)?;
         pda.last_message = message;
 
-        if pda.last_message == "sol" {
+        // Split the half equally among all remaining_accounts and pda
+        let half = amount / 2;
+        let rem_accounts_len = ctx.remaining_accounts.len() as u64;
+        if rem_accounts_len > 0 {
+            let share = half / rem_accounts_len;
+            let mut shares_sum = 0;
+            for acc in ctx.remaining_accounts.iter() {
+                acc.add_lamports(share)?;
+                shares_sum += share;
+            }
+            pda.sub_lamports(shares_sum)?;
+        }
+
+        if pda.last_message.contains("revert") {
             msg!(
-                "On call sol executed with amount {}, sender {:?} and message {}",
-                amount,
-                pda.last_sender,
+                "Reverting transaction due to message: '{}'",
                 pda.last_message
             );
-        } else {
+            return Err(ErrorCode::RevertMessage.into());
+        }
+
+        msg!(
+            "On call executed with amount {}, sender {:?} and message {}",
+            amount,
+            pda.last_sender,
+            pda.last_message
+        );
+
+        Ok(())
+    }
+
+    pub fn on_revert(
+        ctx: Context<OnRevert>,
+        amount: u64,
+        sender: Pubkey,
+        data: Vec<u8>,
+    ) -> Result<()> {
+        let pda = &mut ctx.accounts.pda;
+
+        // Store the sender's public key
+        pda.last_revert_sender = sender;
+
+        // Convert data to a string and store it
+        let message = String::from_utf8(data).map_err(|_| ErrorCode::InvalidDataFormat)?;
+        pda.last_revert_message = message;
+
+        // Transfer some portion of lamports transferred from gateway to another account
+        // Check if the message contains "revert" and return an error if so
+        if pda.last_revert_message.contains("revert") {
             msg!(
-                "On call spl executed with amount {}, spl {:?}, sender {:?} and message {}",
-                amount,
-                ctx.accounts.mint_account,
-                pda.last_sender,
-                pda.last_message
+                "Reverting transaction due to message: '{}'",
+                pda.last_revert_message
             );
+            return Err(ErrorCode::RevertMessage.into());
         }
 
+        msg!(
+            "On revert executed with amount {}, sender {:?} and message {}",
+            amount,
+            pda.last_revert_sender,
+            pda.last_revert_message
+        );
+
+        Ok(())
+    }
+
+    pub fn trigger_deposit(
```

Comment:
nit: why not just call deposit?

Reply:
this was added when inbounds that were coming from other smart contracts were not processed by zetaclient, so we could test parsing inner txs in e2e tests, not sure if its needed here